prime-dotnet-webapi/Services/HealthAuthorityService.cs

```diff
@@ -214,5 +226,33 @@ public async Task<bool> VendorExistsOnHealthAuthorityAsync(int healthAuthorityId
                 .AnyAsync(vendor => vendor.Id == healthAuthorityVendorId
                     && vendor.HealthAuthorityOrganizationId == healthAuthorityId);
         }
+
+        public async Task<bool> IsVendorInUseAsync(int healthAuthorityId, int healthAuthorityVendorCode)
```

Comment:
Names are important because often they can be misleading if not carefully chosen.
This method is called by hitting the API at `api/health-authorities/1/vendors/5`. That "5" isn't a `VendorCode`, its a _`HealthAuthorityVendorId`_. Some confusion may have been caused due to the equivalent PUT request to `api/health-authorities/1/vendors` taking a list of `VendorCodes` to create the `HealthAuthorityVEndor`s.

So if this parameter is renamed to `HealthAuthorityVendorId` its clear the method can be simplified:

```C#
return await _context.HealthAuthoritySites
    .AsNoTracking()
    .AnyAsync(site => site.HealthAuthorityOrganizationId == healthAuthorityId
        && site.HealthAuthorityVendorId == healthAuthorityVendorId);
```

Reply:
Hmm, on the front end, I only have access to the Vendor's code, careSettingCode and name. That's why I am sending the code and getting the ID on the backend. Attached is a screenshot of the info I have for a vendor on the vendor page. 
![image](https://user-images.githubusercontent.com/34261366/145128886-17a105a7-2184-48cb-a9a7-b76d61e8f972.png)

Reply:
Found a way to get the `vendorIds` on that page. Working on getting those ids

Reply:
Refactored and now we have the ids of care-types and vendors on their respective pages.